### ISSUE TITLE
Destructuring a value off of a stateless function causes false prop-types error

### ISSUE THREAD

**Issue by kentcdodds** [yannickcr/eslint-plugin-react#782]
As noted in #735: 

Here's how I can reproduce the issue:

``` javascript
import React, {PropTypes} from 'react'

function Foo({bar}) { // react/prop-types error reported for bar on this line
  const {baz} = Foo
  return <div>{baz} {bar}</div>
}

Foo.propTypes = {
  bar: PropTypes.string.isRequired,
}

Foo.baz = 'hi'
```

This does not produce any error:

``` javascript
import React, {PropTypes} from 'react'

function Foo({bar}) {
  const baz = Foo.baz
  return <div>{baz} {bar}</div>
}

Foo.propTypes = {
  bar: PropTypes.string.isRequired,
}

Foo.baz = 'hi'
```

[Here's a real use case](https://github.com/javascriptair/site/blob/30b5930230bf8cb98ce21c04d21afac72d0bd612/src/pages/episode/sections/show-notes/person-notes.js#L70-L100)

Looks like destructuring a property off of the function (whether inside the function itself or not) is causing issues.
